lib/jazzy/sourcekitten.rb

```diff
@@ -584,8 +584,8 @@ def self.mergeable_objc?(decl, root_decls)
 
     # Two declarations get merged if they have the same deduplication key.
     def self.deduplication_key(decl, root_decls)
-      if decl.type.swift_extensible? || decl.type.swift_extension?
```

Comment:
We can't delete this first 'if' because it breaks merging Swift extensions with Swift types.

I think you've spotted the key though, which is that we can identify Swift extensions of ObjC types from the USR of the extension -- it begins with `c:objc`.  So you should be able to write a `SourceDeclaration::swift_objc_extension?` and use that to decide whether to use the objc dedup key (your new if) or the swift dedup code (the old if).